### ISSUE TITLE
Allow boot scripts to return promise instead of accepting a callback arg

### ISSUE THREAD

**Issue by bajtos** [strongloop/loopback-boot#192]
I'd like to write boot scripts using Promise-based API provided by loopback models:

``` js
module.exports = function(app) {
  return app.models.Product.create({ name: 'pen' });
};
```

**Comment by raabbajam** [comment]
Great idea.

In the meantime here is how to work with Promise and callback:
```
const Promise = require('bluebird'); // make sure to use bluebird
module.exports = function (server, done) {
  const {cookie: cookieModel, Account} = server.models;

  // do all kind of promise things
  return Promise.all([
    cookieModel.find({fields: ['id', 'accountId']}),
    Account.find({fields: ['id']}),
  ])
    .spread((cookies, accounts) => {
      const accountIds = accounts.map(({id}) => id);
      const orphanedCookies = cookies.filter(({accountId = 0}) =>
        !accountIds.includes(accountId));
      return orphanedCookies;
    })
    .map((cookie) => cookie.destroy())
    .asCallback(done); // <------- add asCallback(done)
};

```

**Comment by bajtos** [comment]
I think this feature should be very easy to implement and therefore it's a great opportunity for first-time contributors to this project. Read http://loopback.io/doc/en/contrib/code-contrib.html to get started, I am happy to help you along the way.

**Comment by bajtos** [comment]
@joshuawootonn thank you for the offer! I checked the source code and apparently this feature has been already implemented - see #253 (master branch/version 3.x) and #252 (version 2.x).
